localstack-core/localstack/services/cloudformation/engine/template_deployer.py

```diff
@@ -756,8 +753,10 @@ def _resolve_refs_recursively(
                     {inner_list[0]: inner_list[1]},
                 )
 
+        # remove _aws_no_value_ from resulting references
+        clean_list = []
```

Comment:
Thank you for writing the longer (but easier to understand) loop manually building up a new list, rather than some horribly complex list comprehension!